cloudinit/sources/helpers/vultr.py

```diff
@@ -32,7 +32,7 @@ def get_metadata(
                 connectivity_url_data={"url": url},
             ):
                 # Fetch the metadata
-                v1 = read_metadata(url, timeout, retries, sec_between, agent)
+                v1 = _read_metadata(url, timeout, retries, sec_between, agent)
```

Comment:
>  I prefer to convert from bytes to strings ASAP

generally no objections to this philosophy, however in cases like this co-locating the `.decode()` with the `json.loads()` could make the exception handling simpler by including the exception handler that's already required for the `json.loads()` with the `UnicodeDecodeError` handler. That said, we aren't actually handling this exception here, so that's not an argument for keeping it this way. I can drop the changes to this file if you want.

Reply:
> I can drop the changes to this file if you want.

@TheRealFalcon ping - that whole commit might be unwanted given the philosophy described above. Thoughts on how to proceed?

Reply:
It was a nit, so it's not a strong opinion. I'm fine keeping it this way.